src/rest-server/src/models/job.js

```diff
@@ -395,7 +453,15 @@ class Job {
         appCompletedTime: frameworkStatus.applicationCompletedTimestamp,
         appExitCode: frameworkStatus.applicationExitCode,
```

Comment:
can we make the whole exit schema like below?

        appExitSpec:
        appExitDiagnostics: {
          contaierStderr:
          runtimeError:
          launcherError: 
        },
        appExitTriggerMessage: frameworkStatus.applicationExitTriggerMessage,
        appExitTriggerTaskRoleName: frameworkStatus.applicationExitTriggerTaskRoleName,
        appExitTriggerTaskRoleIndex: frameworkStatus.applicationExitTriggerTaskRoleIndex,

Reply:
The appExitSpec is appStaticExitInfo.
The appExitDiagnostics.launcher is:
applicationExitDiagnostics - contaierStderr - runtimeError

---
In reply to: [276916876](https://github.com/Microsoft/pai/pull/2550#discussion_r276916876) [](ancestors = 276916876)

Reply:
Removed fields, such as appExitDiagnostics and appExitType, can be kept for backward compatiable.
But should be marked as deprecated, and should be just a copy from appExitSpec and appExitDiagnostics.launcher 



---
In reply to: [276917277](https://github.com/Microsoft/pai/pull/2550#discussion_r276917277) [](ancestors = 276917277,276916876)

Reply:
The display importance should be:

appExitCode > appExitSpec > (runtime msg > container msg > launcher msg) > trigger


---
In reply to: [276916876](https://github.com/Microsoft/pai/pull/2550#discussion_r276916876) [](ancestors = 276916876)